### ISSUE TITLE
No Schema Discovery when using Resource/Data-Source specific templates

### ISSUE THREAD

**Issue by elXaptation** [hashicorp/terraform-plugin-docs#139]
Not sure if I have the wrong expectation or if something is broken with resource and data-source templates. The template line:

{{ .SchemaMarkdown | trimspace }} seems to break/not-function when I create these specific templates.

I have my internal/provider/resource_plugin.go and internal/provider/datasource_plugin.go files defined. I create templates:

templates/resource/plugin.md.tmpl and templates/data-sources/plugin.md.tmpl

run tfplugindocs in the root of my plugin repo and get an error:

`Error executing command: unable to generate website: unable to render template "data-sources/plugin.md.tmpl": unable to execute template: template: docTemplate:11:21: executing "docTemplate" at <trimspace>: invalid value; expected string
`

This trimspace error is related to .SchemaMarkdown return "<no value>". Seems that .SchemaMarkdown can't find the 

`func resourcePlugin() *schema.Resource` in resource_plugin.go (nor datasource_plugin.go). 

If I don't create specific templates, tfplugindocs runs to completion and all the files are generated. Although without the custom content I'd like to include.

I'm expecting my specific templates to allow me to create custom md-headers and other content, while .SchemaMarkdown does it's thing.

Creating the specific template index.md.tmpl works as I expect.

Is this how templates/resource/plugin.md.tmpl and templates/datasources/plugin.md.tmpl are suposed to work? 

Thank you.

**Comment by jacobbednarz** [comment]
this isn't just you; i'm also hitting this. somewhere between 0.5.1 and 0.9.0 something is now causing this individual resource templates to fail. i'm still trying to put together a test case to bisect.

**Comment by jacobbednarz** [comment]
looks like there is more nuance here than "doesn't work with 0.9.0" as https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy works with individual resource templates _and_ is using 0.9.0. still digging...

**Comment by jacobbednarz** [comment]
narrowing this down further; `tfplugindocs` by itself works but `tfplugindocs generate` does not.

**Comment by jacobbednarz** [comment]
here are my findings so far.

to have individual templates rendered you need to:

- use `tfplugindocs` directly, not `tfplugindocs generate`
- have the template filename wthout the provider name prefixed (i.e. `access_application.md.tmpl` not `cloudflare_access_application.md.tmpl` like when you want to copy the file verbatim.

will need to dig into why this is and doesn't meet the same criteria as the `tfplugindocs generate` command as it also means you cannot pass in configuration flags with the default commands.

**Comment by elXaptation** [comment]
Oh. Interesting.  Thanks for sharing. Still wonder if this is intended or
not. Seems as to not be intended.

**Comment by jacobbednarz** [comment]
I wouldn't suspect it is intended. Now that I've found where it works, I can have a look into why this isn't working.

**Comment by jacobbednarz** [comment]
found the cause of this and addressed in #148; blame past Jacob. feel free to pull that locally and give it a whirl.

**Comment by elXaptation** [comment]
Hey Guys, 

Seems something still a miss here. Related to naming, I believe 0.10.0 is unable to find some resource it created itself. This provider doesn't have any actual template defined.. so tfplugindocs creates all the templates, then complains it can't find them.. Any thoughts?

```
elXaptation:~/terraform-provider-plat$ ../tfplugindocs_0100 
rendering website for provider "terraform-provider-plat" (as "terraform-provider-plat")
copying any existing content to tmp dir
exporting schema from Terraform
compiling provider "plat"
using Terraform CLI binary from PATH if available, otherwise downloading latest Terraform CLI binary
running terraform init
getting provider schema
rendering missing docs
generating missing resource content
generating template for "cloud_router_bgp_session"
generating template for "cloud_services_aws_create_backbone_dedicated_cr"
generating template for "cloud_services_aws_hosted_connection"
generating template for "cloud_services_aws_req_hosted_conn"
generating template for "interface"
generating template for "aws_cloud_router_connection"
generating template for "cloud_router"
generating template for "cloud_router_bgp_prefixes"
generating template for "cloud_services_aws_dedicated"
generating template for "cloud_services_aws_hosted_marketplace"
generating template for "cloud_services_aws_provision_requested_mkt_conn"
generating missing data source content
generating template for "cloud_services_aws_connection_info"
generating template for "cloud_services_aws_dedicated_conn"
generating template for "locations"
generating template for "aws_cloud_router_connection"
generating template for "billing"
generating template for "cloud_router"
generating template for "cloud_router_bgp_prefixes"
generating template for "cloud_router_bgp_session"
generating missing provider content
provider "terraform-provider-plat" template exists, skipping
rendering static website
cleaning rendered website dir
rendering templated website to static markdown
rendering "data-sources/aws_cloud_router_connection.md.tmpl"
Error executing command: unable to generate website: unable to find resource for provider (plat) and template (aws_cloud_router_connection.md.tmpl)
```
I verified the md.tmpl files where created in the tmp directory. Any suggestions on how to trouble shoot this further?

**Comment by jacobbednarz** [comment]
can you link me to your repository? I suspect the filename is incorrect and not matching what the schema expects for this resource as the others work just fine.

**Comment by elXaptation** [comment]
This issue is resolved in 0.10.1 #150
